src/apis/auth/index.ts

```diff
@@ -85,10 +87,19 @@ export async function resetPassword({
   return response.data;
 }
 
-export async function checkNickname(nickname: string) {
-  const response = await AuthAxios.get<NicknameCheckResponse>(
-    `/api/auth/check-nickname?nickname=${nickname}`,
+export const checkAvailability = async (type: string, value: string) => {
```

Comment:
```suggestion
export const checkAvailability = async (type: 'email' | 'nickname', value: string) => {
```

type의 타입을 유니온 타입으로 지정하면 좀 더 간편하고 정확하게 사용할 수 있을 것 같은데 어떻게 생각하시나요?

Reply:
피드백 감사합니다 🙇

현재 해당 함수의 `type` 매개변수는 `email`과 `nickname` 두 가지 값만 사용되지만,
추후 API 확장성을 고려해 string 타입으로 지정했었습니다.

말씀해 주신 대로, 유니온 타입도 확장성 면에서 괜찮은 선택이지만,
보다 명확한 타입 정의를 위해 커스텀 타입을 별도로 생성하는 방향으로 수정하였습니다.